### ISSUE TITLE
Media: "add via url" should relax file type requirements

### ISSUE THREAD

**Issue by rralian** [Automattic/wp-calypso#10560]
Many screen-shotting services, like cloudup or droplr, give you a URL that takes you to a webpage that features that image... but you can get a direct link to the image file if you just add a "+" to the end of the URL (example: [http://cld.wthms.co/Hl06+](http://cld.wthms.co/Hl06+)). The problem is, if you use that URL with "add via url," at least in Calypso, we see that the filename doesn't end in a known media extension and we disallow it saying "the file type is not supported."

We should relax this extension checking on the client... at least support this convention of adding a plus to an alphanumeric string as a valid image. Or we should disable the check entirely and let the check happen on the server where we can better confirm the nature of the file.

/cc @gwwar

**Comment by aduth** [comment]
3020570-t

**Comment by rralian** [comment]
I spent a little time on this last night for funzies, since this is a pet peeve of mine. @aduth fwiw I don't have access to that help ticket, so I don't have the context, in case there's some ideas there.

Unfortunately I've found that whitelisting file extensions is pretty embedded into WordPress Core and Jetpack as a security concern. I don't think I'd be able to tackle this without some deeper discussions of security and some fairly heavy surgery in Jetpack, wpcom, calypso, and (maybe?) wp-core.

I also looked into having the client follow the redirect (307 for droplr, 302 for cloudup) and instead use the target URL, but from what I can tell you cannot obtain the target URL of a redirected link on the client.

Wondering if anyone else has other suggestions/ideas? Otherwise I may close this as "wontfix". @westi? @aduth?
